### PR TITLE
Bug fixing and code refine

### PR DESCRIPTION
**Pull request by zhenlinluo**
Hi @piiswrong 

Pls kindly review and approve. This is to address your comments for code and provide small bug fixing. 

Thanks

### CODE REVIEW

File: src/operator/concat.cc [hunk 1]
```diff
@@ -21,11 +21,9 @@ Operator* CreateOp<cpu>(ConcatParam param, int dtype) {
   if (1 == param.dim) {
     switch (dtype) {
       case mshadow::kFloat32:
-      op = new MKLConcatOp<cpu, float>(param);
-      break;
+      return new MKLConcatOp<cpu, float>(param);
```
Comment (piiswrong) [inline src/operator/concat.cc]: could you also replace the slice layer with mkl?

Comment (zhenlinluo) [inline src/operator/concat.cc]: Eric, can you let me know what topology will require slice layer?

Comment (piiswrong) [inline src/operator/concat.cc]: what's topology? You basically swap the forward and backward of concat to get slice

**Comment by piiswrong** [comment]
@zhenlinluo 
The following code gives segfault:
>>> import mxnet as mx
>>> import numpy as np
>>> sym, arg_params, aux_params = mx.model.load_checkpoint('resnet-50', 0)
>>> model = mx.mod.Module(symbol=sym, context=mx.cpu(0))
>>> model.bind(for_training=False, data_shapes=[('data', (1, 3, 300, 400))])
>>> model.set_params(arg_params, aux_params)
>>> model.forward(mx.io.DataBatch(data=[mx.nd.zeros((1,3,300,400))], label=[]), is_train=False)
>>> MKL Build:20161005
Segmentation fault (core dumped)

**Comment by zhenlinluo** [comment]
Eric, do you do "export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/the path to MKL lib"?

**Comment by piiswrong** [comment]
Yes

**Comment by lingyanz** [comment]
I download latest mxnet code and have a try, without seg fault
could you give me you backtrace
$ gdb python
(gdb) run python.py 
(gdb) bt

**Pull request merged**